File: app/src/main/java/com/mbmc/template/Application.java
package com.mbmc.template;

import com.mbmc.template.di.component.ApplicationComponent;
import com.mbmc.template.di.component.DaggerApplicationComponent;
import com.mbmc.template.di.component.UiComponent;
import com.mbmc.template.di.module.MainModule;


public class Application extends android.app.Application {

    private ApplicationComponent applicationComponent;


    @Override
    public void onCreate() {
        super.onCreate();

        applicationComponent = DaggerApplicationComponent.builder()
                .mainModule(new MainModule(this)).build();
    }

    public UiComponent getUiComponent() {
        return applicationComponent.uiComponent();
    }

}

File: core/src/main/kotlin/com/mbmc/template/core/Application.kt
package com.mbmc.template.core

import com.mbmc.template.core.di.component.CoreComponent
import com.mbmc.template.core.di.component.CoreComponentProvider
import com.mbmc.template.core.di.component.DaggerCoreComponent

open class Application : android.app.Application(), CoreComponentProvider {
    private val _coreComponent: CoreComponent by lazy {
        DaggerCoreComponent.create()
    }

    override fun onCreate() {
        super.onCreate()
    }

    override fun getCoreComponent(): CoreComponent = _coreComponent
}